### PR TITLE
Adding typescript support

### PR DESCRIPTION
**Pull request by Samuel-Gill**
In this PR, I have converted the JavaScript files to TypeScript files. Run the application using nodemon and it is launching fine.
Please review and recommend if any changes are required.

### CODE REVIEW

File: src/ip-selector/default-ip-selector.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { Request } from "express";
+
+const originatingIPSelector = (req: Request): string => {
+  return req.ip;
+};
+
+export default originatingIPSelector;
+
```
Comment (UmanShahzad) [inline src/ip-selector/default-ip-selector.ts:8]: Rm trailing newline in file

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import originatingIPSelector from "./ip-selector/originating-ip-selector";
+
+type MiddlewareOptions = {
+    token?: string;
+    cache?: any;
+    timeout?: number;
+    ipSelector?: (req: any) => string;
+};
+  
+const ipInfoMiddleware = ({ token = "", cache, timeout, ipSelector }: MiddlewareOptions = {}) => {
```
Comment (UmanShahzad) [inline src/index.ts:12]: Name it `ipinfo` instead of `ipInfo` in the code.

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    const ipinfo = new IPinfoWrapper(token, cache, timeout);
+  if (ipSelector == null || typeof ipSelector !== "function") {
```
Comment (UmanShahzad) [inline src/index.ts:14]: The spacing is messed up here. Maybe we can add `prettier` and stabilize things. Then we can make spacing 4 as well.

File: package.json [hunk 4]
```diff
@@ -21,10 +21,14 @@
         "node": ">=10"
     },
     "dependencies": {
+        "@types/node": "^18.15.11",
+        "ipinfo-express": "^1.1.0",
```
Comment (UmanShahzad) [inline package.json:25]: We're putting our own package as a dependency?

File: package.json [hunk 5]
```diff
@@ -21,10 +21,14 @@
...
         "node-ipinfo": "^3.0.2"
     },
     "devDependencies": {
+        "@types/express": "^4.17.17",
         "dotenv": "^8.2.0",
-        "prettier": "^2.2.1"
+        "prettier": "^2.2.1",
```
Comment (UmanShahzad) [inline package.json:31]: Oh we already have prettier installed... then we need to run it. See `scripts/fmt.sh` in this repo.

File: ipinfo-express.d.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      loc: string;
+      org: string;
+      postal: string;
+      timezone: string;
+    }
+  
+    function ipinfo(options: IPInfoMiddlewareOptions): RequestHandler;
+  
+    export = ipinfo;
+  }
```
Comment (UmanShahzad) [inline ipinfo-express.d.ts:26]: Spacing is messed up

File: ipinfo-express.d.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+declare module "ipinfo-express" {
+    import { RequestHandler } from "express";
+  
+    interface IPInfoMiddlewareOptions {
```
Comment (UmanShahzad) [inline ipinfo-express.d.ts:4]: When in caps mode, spelling is `IPinfo` not `IPInfo`

**Review by UmanShahzad — CHANGES_REQUESTED** [review]

**Comment by Samuel-Gill** [comment]
@UmanShahzad Sure thing. I will apply the suggested changes and ping you again. Thanks for review.

**Comment by Samuel-Gill** [comment]
@UmanShahzad Please review changes.

File: ipinfo-express.d.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    interface IPinfoResponse {
+        ip: string;
+        hostname: string;
+        city: string;
+        region: string;
+        country: string;
+        loc: string;
+        org: string;
+        postal: string;
+        timezone: string;
```
Comment (UmanShahzad) [inline ipinfo-express.d.ts:20]: Sorry one more thing - this actually is a much bigger object if we use a higher-permissions token. @rm-Umar can you point @Samuel-Gill to the Nodejs SDK where this exists already? @Samuel-Gill do take a look there too.

Comment (rm-Umar) [reply ipinfo-express.d.ts:20]: @Samuel-Gill check this out https://github.com/ipinfo/node/blob/master/src/common.ts

Comment (Samuel-Gill) [reply ipinfo-express.d.ts:87]: Updated.

**Review by UmanShahzad — APPROVED** [review]

**Comment by UmanShahzad** [comment]
@rm-Umar could you show @Samuel-Gill how we upload this as well?

**Comment by UmanShahzad** [comment]
(need a version bump too I suppose!)

**Comment by Samuel-Gill** [comment]
> @rm-Umar could you show @Samuel-Gill how we upload this as well?

published.

**Review by UmanShahzad — COMMENTED** [review]
The push that was done from this PR resulted in a fully-TS push which breaks all non-TS users. Presumably broken push done by @Samuel-Gill & @rm-Umar 

https://github.com/ipinfo/node-express/issues/13

**Pull request merged**